### PR TITLE
github-merge: Warn when commit message has HTML comment

### PR DESCRIPTION
**Pull request by MarcoFalke**

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Can be tested by merging https://github.com/bitcoin/bitcoin/pull/18380

**Comment by laanwj** [comment]
ACK 1a9320263b3943dac09e7471609fbd571ac7dd9d, thanks

File: github-merge.py [hunk 1]
```diff
@@ -200,6 +200,8 @@ def print_merge_details(pull, title, branch, base_branch, head_branch, acks, mes
             print('{}Top commit has no ACKs!{}'.format(ATTR_WARN, ATTR_RESET))
     if message is not None and '@' in message:
         print('{}Merge message contains an @!{}'.format(ATTR_WARN, ATTR_RESET))
+    if message is not None and '<!-' in message:
+        print('{}Merge message contains an html comment!{}'.format(ATTR_WARN, ATTR_RESET))
```
Comment (laanwj) [inline github-merge.py]: Looking at this more closely, this seems incorrect. `message` is overwritten by the `acks` loop above. Due to Python's lack of varilable scopring it contains the last ACK message.

Comment (MarcoFalke) [reply github-merge.py]: I saw that too, but I think this was tested for `@` (works the same)

And 18380 can be used to test the html comment

Comment (MarcoFalke) [reply github-merge.py]: Hmmm

```python
>>> def fun(i):
...  print(i)
...  for i in range(2):
...   print(i)
...  print(i)
... 
>>> fun(333)
333
0
1
1

Comment (MarcoFalke) [reply github-merge.py]: Oh, it works because it has no ACKs :man_facepalming:

**Comment by laanwj** [comment]
LGTM now, re-ACK 9320f155f71f8f374442206027d3d440d3725bd2

**Comment by jonatack** [comment]
This changed the exact same code that #51 does :man_facepalming:

**Pull request merged**